pkg/roachprod/install/cockroach.go

```diff
@@ -405,9 +381,15 @@ func (c *SyncedCluster) Start(ctx context.Context, l *logger.Logger, startOpts S
 	}
 
 	if c.allowServiceRegistration() {
```

Comment:
Can we not move this logic to allowServiceRegistration?
`c.IsLocal()` check is already present. The else block errors out if customPortsSpecified(). The only condition is `startOpts.Target == StartDefault`, which is a noop am can be added as one extra `else if startOpts.Target == StartDefault`

Reply:
It can probably be simplified, but there is a slight difference in intention between `allowServiceRegistration` and the new logic deciding whether it's necessary to register the service. If we were to update the `allowServiceRegistration` function to combine the logic for cases where it's also not necessary we would need the else condition to not print the warning that service discovery is disabled.